windows/src/global/delphi/general/Keyman.System.SettingsManagerFile.pas

```diff
@@ -0,0 +1,101 @@
+unit Keyman.System.SettingsManagerFile;
+
+interface
+
+uses
+  Keyman.System.Settings;
+
+type
+  TKeymanSettingsManagerFile = class
+    ///  <summary>Exports non-default settings to a json settings file.</summary>
+    class procedure Export(Settings: TKeymanSettings; const Filename: string); static;
+
+    ///  <summary>Imports new settings from a json settings file. Settings
+    ///  not present in the file are left unmodified.</summary>
+    class procedure Import(Settings: TKeymanSettings; const Filename: string);
+  end;
+
+implementation
+
+uses
+  System.Classes,
+  System.JSON,
+  System.SysUtils;
+
+{ TKeymanSettingsManager }
+
+class procedure TKeymanSettingsManagerFile.Import(Settings: TKeymanSettings;
+  const Filename: string);
+var
+  ID: string;
+  s: TKeymanSetting;
+  json: TJSONObject;
+  stream: TStringStream;
+  p: TJSONPair;
+begin
+  stream := TStringStream.Create('', TEncoding.UTF8);
+  try
+    stream.LoadFromFile(Filename);
+    json := TJSONObject.ParseJSONValue(stream.DataString, False, True) as TJSONObject;
```

Comment:
Are there any warnings if importing invalid JSON?

Reply:
Invalid JSON causes an exception which will be handled by the caller